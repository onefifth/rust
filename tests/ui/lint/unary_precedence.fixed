//@ check-fail
//@ run-rustfix

fn main() {
    let _ = -(1i32.abs());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f32.abs());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f64.asin());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f64.asinh());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f64.tan());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f64.tanh());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1.0_f64.cos().cos());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1.0_f64.cos().sin());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1.0_f64.sin().cos());
    //~^ ERROR unary operator `-` has lower precedence than method call
    let _ = -(1f64.sin().sin());
    //~^ ERROR unary operator `-` has lower precedence than method call

    dbg!( -(1.0_f32.cos()) );
    //~^ ERROR unary operator `-` has lower precedence than method call

    // should not warn
    let _ = (-1i32).abs();
    let _ = (-1f32).abs();
    let _ = -(1i32).abs();
    let _ = -(1f32).abs();
    let _ = -(1i32.abs());
    let _ = -(1f32.abs());
}
