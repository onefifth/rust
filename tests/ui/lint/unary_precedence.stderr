error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:5:13
   |
LL |     let _ = -1i32.abs();
   |             ^^^^^^^^^^^
   |
   = note: `#[deny(ambiguous_unary_precedence)]` on by default
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1i32.abs());
   |              +          +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:7:13
   |
LL |     let _ = -1f32.abs();
   |             ^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f32.abs());
   |              +          +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:9:13
   |
LL |     let _ = -1f64.asin();
   |             ^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f64.asin());
   |              +           +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:11:13
   |
LL |     let _ = -1f64.asinh();
   |             ^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f64.asinh());
   |              +            +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:13:13
   |
LL |     let _ = -1f64.tan();
   |             ^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f64.tan());
   |              +          +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:15:13
   |
LL |     let _ = -1f64.tanh();
   |             ^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f64.tanh());
   |              +           +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:17:13
   |
LL |     let _ = -1.0_f64.cos().cos();
   |             ^^^^^^^^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1.0_f64.cos().cos());
   |              +                   +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:19:13
   |
LL |     let _ = -1.0_f64.cos().sin();
   |             ^^^^^^^^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1.0_f64.cos().sin());
   |              +                   +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:21:13
   |
LL |     let _ = -1.0_f64.sin().cos();
   |             ^^^^^^^^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1.0_f64.sin().cos());
   |              +                   +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:23:13
   |
LL |     let _ = -1f64.sin().sin();
   |             ^^^^^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     let _ = -(1f64.sin().sin());
   |              +                +

error: unary operator `-` has lower precedence than method calls which might be unexpected
  --> $DIR/unary_precedence.rs:26:11
   |
LL |     dbg!( -1.0_f32.cos() );
   |           ^^^^^^^^^^^^^^
   |
help: use parentheses to clarify your intent
   |
LL |     dbg!( -(1.0_f32.cos()) );
   |            +             +

error: aborting due to 11 previous errors

